src/vs/code/electron-browser/sharedProcess/sharedProcessMain.ts

```diff
@@ -204,6 +207,10 @@ class SharedProcessMain extends Disposable {
 		const diskFileSystemProvider = this._register(new DiskFileSystemProvider(logService));
 		fileService.registerProvider(Schemas.file, diskFileSystemProvider);
 
+		const localFileSystemProviderFromMain = this._register(new IPCFileSystemProvider(mainProcessService.getChannel('localFilesystem'), { pathCaseSensitive: isLinux }));
```

Comment:
After this is merged I can see how to make this more reusable, i.e. have some factory method.